kinesis/dynamodb.py

```
@@ -0,0 +1,383 @@
+
+                # Try to get existing table
+                try:
+                    self._table = await dynamodb.Table(self.table_name)
+                    await self._table.load()
+                    log.info(f"Using existing DynamoDB table: {self.table_name}")
+                except ClientError as e:
+                    if e.response["Error"]["Code"] == "ResourceNotFoundException":
+                        if self.create_table:
+                            await self._create_table(dynamodb)
+                        else:
+                            raise Exception(f"DynamoDB table {self.table_name} does not exist")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)